src/Mapbender/CoreBundle/Controller/ApplicationController.php

```diff
@@ -393,7 +400,28 @@ public function instanceTunnelAction($slug, $instanceId)
                 $url = $source->getGetFeatureInfo()->getHttpGet();
                 break;
             case 'getlegendgraphic':
-                $url = $source->getGetLegendGraphic()->getHttpGet();
+                $glgMode = $request->query->get('_glgmode', null);
+                $layerName = $request->query->get('layer', null);
+                if (!$layerName) {
+                    $glgMode = null;
+                    $layerSource = null;
+                } else {
+                    $layerSource = WmsSourceEntityHandler::getLayerSource($source, $layerName);
+                    if (!$layerSource) {
+                        $glgMode = null;
+                    }
+                }
+                switch ($glgMode) {
```

Comment:
I prefer to create a new method for: `getUrlByMode($mode)`

Reply:
Agreed. Will do.